Lib9c/Action/Staking.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType("staking")]
+    public class Staking : GameAction
+    {
+        public int level;
+        public int stakingRound;
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            IAccountStateDelta states = context.PreviousStates;
+            Address stakingAddress = StakingState.DeriveAddress(context.Signer, stakingRound);
+            if (context.Rehearsal)
+            {
+                return states
+                    .SetState(stakingAddress, MarkChanged)
+                    .SetState(context.Signer, MarkChanged)
+                    .MarkBalanceChanged(GoldCurrencyMock, context.Signer, stakingAddress);
+            }
+
+            StakingSheet stakingSheet = states.GetSheet<StakingSheet>();
+
+            AgentState agentState = states.GetAgentState(context.Signer);
+            if (agentState is null)
+            {
+                throw new FailedLoadStateException("Aborted as the agent state failed to load.");
+            }
+
+            if (agentState.StakingRound != stakingRound)
+            {
+                throw new InvalidStakingRoundException(
+                    $"Expected staking round is {agentState.StakingRound}, but actual staking round is {stakingRound}.");
+            }
+
+            if (!stakingSheet.TryGetValue(level, out StakingSheet.Row _))
+            {
+                throw new SheetRowNotFoundException(nameof(StakingSheet), level);
+            }
+
+            Currency currency = states.GetGoldCurrency();
+            FungibleAssetValue requiredGold = currency * 0;
```

Comment:
Is this zero intended? (if so, it would be better leaving comment for justifying)

Reply:
default value is intended. I will leave it as a comment.